src/opm/io/eclipse/ESmry.cpp

```diff
@@ -426,13 +426,13 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData) :
             for (size_t i=0; i < keywords.size(); i++) {
                 Opm::EclIO::lgr_info lgr { lgrs[i], {numlx[i], numly[i], numlz[i]}};
                 const std::string keyw = makeKeyString(keywords[i], wgnames[i], nums[i], lgr);
-                if (keywList.find(keyw) != keywList.end())
+                if ((!keyw.empty() > 0) && (keywList.find(keyw) != keywList.end()))
```

Comment:
uhm.

Reply:
Summary files generated by the Eclipse simulator creates a lot of time vectors which are not real data. These should be ignored by ESmry. The member function ESmry::makeKeyString will return an empty string for this entries and importantly arrayPos should not be updated with this vectors. 

Without this fix, arrayPos with key equal to 0 will be set to wrong array position. I have a small summary file generated by Eclipse which I can add as a test for this if you like.

Reply:
sure. but make up your mind no using .empty() or size() > 0, this hybrid won't work well.

Reply:
Sorry, my bad. Should be fixed now.